components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+                     self._cronGenerator.find("#dayWeek").val());
+                 var expression = self._cronGenerator.find("#expression").val();
+                 var description = Cronstrue.toString(expression, { throwExceptionOnParseError: false });
+
+                 var weekValue = self._cronGenerator.find('#dayWeek').val();
+                 var dayMonValue = self._cronGenerator.find('#dayMonth').val();
+                 if(isValidDayOfWeekValue(weekValue, dayMonValue)){
+                      self._cronGenerator.find('#dayWeek').removeClass("error-field");
+                      self._cronGenerator.find("#output").text(description);
+                      self._cronGenerator.find('#errorMsg').hide();
+                 } else {
+                      self._cronGenerator.find('#dayWeek').addClass("error-field");
+                      self._cronGenerator.find('#errorMsg').text(errorMsg);
+                      self._cronGenerator.find('#errorMsg').show();
+                 }
+                 self.isErrorOccured();
+            });
+            self._cronGenerator.find("#expression").on("input",function(){
+                 var expression = self._cronGenerator.find("#expression").val();
+                 var description = Cronstrue.toString(expression, { throwExceptionOnParseError: false });
+                 var splitValues = expression.split(" ");
+                 var second = splitValues[0];
+                 var minute = splitValues[1];
+                 var hour = splitValues[2];
+                 var dayMonth = splitValues[3];
+                 var month = splitValues[4];
+                 var dayWeek = splitValues[5];
+                 if(isValidCronExpression(expression) === true){
+                     self._cronGenerator.find('#expression').removeClass("error-field");
+                     self._cronGenerator.find("#output").text(description);
+                     self._cronGenerator.find('#output').removeClass("error-message");
+                     self._cronGenerator.find('#saveButton').removeAttr("disabled");
+                     self._cronGenerator.find("#second").val(splitValues[0]);
+                     self._cronGenerator.find("#minute").val(splitValues[1]);
+                     self._cronGenerator.find("#hour").val(splitValues[2]);
+                     self._cronGenerator.find("#dayMonth").val(splitValues[3]);
+                     self._cronGenerator.find("#month").val(splitValues[4]);
+                     self._cronGenerator.find("#dayWeek").val(splitValues[5]);
+                 } else{
+                    self._cronGenerator.find('#expression').addClass("error-field");
+                    self._cronGenerator.find('#output').html("Invalid Cron Expression");
+                    self._cronGenerator.find('#output').addClass("error-message");
+                    self._cronGenerator.find('#saveButton').attr("disabled","disabled");
+                 }
+                 if(isValidTimeValue(second, 59)){
```

Comment:
when you are going to validate time of seconds why you need to pass a static value 59 here ?

Reply:
fixed in 0fb87f3